sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -52,21 +52,36 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     }
   }
 
+  private def cache(relation: DataSourceV2Relation, cacheInfo: CacheInfo): Unit = {
+    session.sharedState.cacheManager.cacheQuery(
+      session,
+      relation,
+      cacheInfo.tableName,
+      cacheInfo.storageLevel)
+  }
+
   private def refreshCache(r: DataSourceV2Relation)(): Unit = {
     session.sharedState.cacheManager.recacheByPlan(session, r)
   }
 
-  private def invalidateCache(r: ResolvedTable, recacheTable: Boolean = false)(): Unit = {
+  // Invalidates the cache associated with the given table. If there exists a cache with the given
+  // table, the cache's info (table name and storage level) is returned.
+  private def invalidateCache(
+      r: ResolvedTable,
+      recacheTable: Boolean = false)(): Option[CacheInfo] = {
```

Comment:
Not sure if the `CacheInfo` class is really needed since here only the storage level returned is used. `Option[StorageLevel]` should work too.

Reply:
+1 for @sunchao 's comment. A simpler one is better.

Reply:
OK, I changed to use `Option[StorageLevel]`. The reason I wrapped it with `CacheInfo` was to simplify callback signatures:
```scala
invalidateCache: () => Option[CacheInfo],
cacheTable: (DataSourceV2Relation, CacheInfo) => Unit
```
vs.
```scala
invalidateCache: () => Option[StorageLevel],
cacheTable: (DataSourceV2Relation, String, StorageLevel) => Unit)
```